Switch from vee-validate to Vuelidate

hubudibu opened the pull request:
Fixes #334 Switch from [vee-validate](http://vee-validate.logaretm.com/) to [vuelidate](https://monterail.github.io/vuelidate/)

affected pages:
- SignUp
- Login Modal
- Edit Profile

The most interesting one is the SignUp page, where I used vuelidate's [async validator](https://monterail.github.io/vuelidate/#sub-asynchronous-validation) feature for checking name availability.

**Ready for review**
_it's branched off #325 New Group Creation, so it will be easier to review after that, but the commits starting from jan 11 are basically about this one_

hubudibu committed:
signup field validators

hubudibu committed:
moving stuff under form.

hubudibu committed:
name availability validator started

hubudibu committed:
name availability async validator

hubudibu committed:
signup validation cleanup

hubudibu committed:
veelidate on login

hubudibu committed:
vuelidate on profile edit page

hubudibu committed:
remove vee-validate

taoeffect commented:
OK, so I had some minor issues with #325, but I decided to merge it because this one (supposedly) builds off of it, and to make your life easier and not screw around with git complexity, I figured the issues that I mentioned in my review comments on #325 can be addressed in this pull request instead.

I posted [6 comments, starting here](https://github.com/okTurtles/group-income-simple/pull/325#pullrequestreview-88651386).

- Please have a look at those and add the changes to this pull request instead of the other one (since that's already merged)
- Also, I'm noticing that for whatever reason, this pull request still shows all of the comments and files as apparently unmerged with respect to `okTurtles:master`. Which is strange since it should show only the changes between this and #325. See if you can figure out why that is (perhaps you have to first merge the changes in `okTurtles:master` into your local branch, and then apply these changes on top of it)

Finally, I didn't get a chance to actually test the new group creation flow, so I'll leave that part for tomorrow and will post any issues I find to this pull request.

Great job so far! 😄 

At least from looking at the code (without actually testing it yet), #325 looked fantastic! 🎉

hubudibu committed:
Merge remote-tracking branch 'upstream/master'

* upstream/master: (68 commits)
  bigger back button
  tests updated to skip rule setting
  circle sliders with majority warning
  install circle slider for rule percentage setting
  dynamic currency list
  save mincome currency
  currency dropdown
  i18n on summary page
  advanced validation parameters
  disable next button if step is invalid
  vuelidate all steps for required
  try vuelidate for group name validation
  install vuelidate for validation
  v-else fix
  statusicon is no  longer used
  v-else
  simplify click handlers
  remove accidental wait
  remove .prevent from buttons as this is not a form element any more
  stepAssistant only needs route names
  ...

hubudibu commented:
So after you merged #325, I could 
- merge upstream master in mine
- rebase this branch on my master
- and push -f it back here so that the confusing history part disappeared 

And now on to the review comments! 🎉

hubudibu committed:
computed currency symbol

hubudibu committed:
mincome accepts decimals w custom validator

hubudibu committed:
no one invited

hubudibu committed:
i18n on majority warning

hubudibu commented:
I addressed all the comments on #325, everything is updated except for [the summary page warnings, see this comment](https://github.com/okTurtles/group-income-simple/pull/325#discussion_r161559956)

I also noticed another bug, that the minimum income field didn't accept decimals, which is not that useful for a mincome set in bitcoin :) now it accepts 2 decimals, based on validation I found on the profile page. let me know if I should increase it.

hubudibu committed:
update centering based on #220

taoeffect commented:
> I also noticed another bug, that the minimum income field didn't accept decimals, which is not that useful for a mincome set in bitcoin :) now it accepts 2 decimals, based on validation I found on the profile page. let me know if I should increase it.

Well, in the future I'm guessing the standard BTC unit will be mBTC or something like that, which is easier to deal with. We might want to have "currency" be a class of some sort that includes various info about the currency like:

- Its full name (Bitcoin)
- Its abbreviated name (BTC)
- Its currency symbol (₿ - yes, [there's now a real](https://en.bitcoin.it/wiki/Bitcoin_symbol), dedicated symbol for Bitcoin in Unicode)
- The number of decimal places allowed (e.g. dollars usually allow for two, whereas BTC requires way more, and mBTC can deal with fewer)

taoeffect commented on frontend/simple/views/SignUp.vue:
```diff
@@ -127,47 +166,53 @@ export default {
+          // async validator can return a promise
+          // we need the opposite of namespace.lookup(value) here
+          return new Promise((resolve, reject) => {
+            namespace.lookup(value).then(reject, resolve)
```
Wow, this is very clean, nicely done.

taoeffect commented:
So this pull request seems ready to go! Great job! Everything seems to be working correctly! 😄 👍 

The only suggestion I have is changing the title of "Group Advanced Rules" to "Voting Rules". But maybe let's leave that for a future PR? This seems ready to be merged as-is now. 🎉
